src/_posts/languages/php/2000-01-01-extensions.md

```diff
@@ -14,57 +14,195 @@ Some of them are pre-installed and shipped by default with the deployed version
+| `ext-xmlwriter`    | Enabled  | Enabled  | Enabled  | Enabled  |
+| `ext-xsl`          | Enabled  | Enabled  | Enabled  | Enabled  |
+| `ext-zend-opcache` | Enabled  | Enabled  | Enabled  | Enabled  |
+| `ext-zip`          | Enabled  | Enabled  | Enabled  | Enabled  |
+| `ext-zlib`         | Enabled  | Enabled  | Enabled  | Enabled  |
+
+### Enabling a PHP Built-in Extension
+
+To enable an optional PHP built-in extension, add its name to the `require`
+block of your `composer.json` file, like so:
+
+```json
+{
+  "require": {
+    "ext-pdo_pgsql": "*",
+    "ext-gettext": "*",
+    ...
+  }
+}
+```
+
+### Adding a PHP Built-in Extension
+
+To add a PHP built-in extension that is not available, please get in touch
+with our support team.
+
 
 ## Working With PECL Extensions
 
-All [PECL extensions](https://pecl.php.net/) are available. If the extension
-you need is not in the above list of pre-installed extensions, it will be
-compiled during the *build* phase of your deployment.
+{% note %}
+Some PECL extensions are very old, probably even unmaintained and will only
+work with an old version of PHP (e.g. `mongo`, `sodium`, `ds`, `lua` or
+`imap`). We strongly advise you to pick your extension(s) carefully.
+{% endnote %}
+
+### Available PECL Extensions
+
+The following table lists the PECL extensions that we pre-compile and make
+**available** on Scalingo.
+
+- Some are enabled by default and can't be disabled. You don't have to add them
+  to your `composer.json` file to enable them (but you can).\
+  They are marked as **Enabled** in the following table.
+- Some others are disabled by default and [can be enabled](#enabling-a-pecl-extension).\
+  They are marked as **Optional** in the following table.
+- Unavailable extensions are marked with a blank cell.
+
+{% note %}
+To use a PECL extension that is not listed in the table of available PECL
+extensions, please read [Adding a PECL Extension](#adding-a-pecl-extension).
+{% endnote %}
+
+| Extension       | PHP 8.1  | PHP 8.2  | PHP 8.3  | PHP 8.4  |
+| --------------- | -------- | -------- | -------- | -------- |
+| `ext-amqp`      | Optional | Optional | Optional | Optional |
+| `ext-apcu`      | Enabled  | Enabled  | Enabled  | Enabled  |
+| `ext-apfd`      | Optional | Optional | Optional | Optional |
+| `ext-event`     | Optional | Optional | Optional | Optional |
+| `ext-igbinary`  | Optional | Optional | Optional | Optional |
+| `ext-imagick`   | Optional | Optional | Optional | Optional |
+| `ext-memcached` | Optional | Optional | Optional | Optional |
+| `ext-mongodb`   | Enabled  | Enabled  | Enabled  | Enabled  |
+| `ext-oci8`      |          | Optional | Optional | Optional |
+| `ext-redis`     | Enabled  | Enabled  | Enabled  | Enabled  |
 
-To enable a native PHP extension, add it in the `require` block of your
-`composer.json`, like so (notice the `ext-` prefix):
+#### Available Versions
+
+The following table lists the supported versions of available PECL extensions:
```

Comment:
question: I don't know if the table is the best here. It looks like all extensions have the same version, whatever the PHP version. Or maybe a table that only lists the versions? But adding a column for each PHP version does not seem really useful, what do you think?

Reply:
I was expecting this question :smiley:

MHO about this:
- Indeed, **currently** all versions are the same. But sometimes it isn't the case, and I guess it won't always be the case. This is a bet on future ^^
- Specifying everything for each supported PHP version allows to remove any doubt (for the record, I've been having doubts while building the tables: "do we _really_ have the same version for PHP 8.1 and 8.4? Or did we miss an update?")
- For extension `oci8` (for example), it felt a bit weird to have a version while it's not available at all for PHP 8.1.
- I agree that it creates big tables with redundant information, and that it will be boring to maintain. But I also think it makes it clearer and more homogeneous, which gets my priority.

WDYT?